### PR TITLE
Quex.offer mod tests

### PR DESCRIPTION
**Pull request by Quexington**
### Purpose:
Adds a bunch of forwards compatibility tests for offers

### CODE REVIEW

File: chia/wallet/nft_wallet/nft_wallet.py [hunk 1]
```diff
@@ -37,7 +37,19 @@
     calculate_synthetic_secret_key,
     puzzle_for_pk,
 )
-from chia.wallet.trading.offer import OFFER_MOD, OFFER_MOD_HASH, NotarizedPayment, Offer
+from chia.wallet.trading.offer import (
+    OFFER_MOD,
+<<<<<<< HEAD
```
Comment (emlowe) [inline chia/wallet/nft_wallet/nft_wallet.py:42]: This seems to have merge conflict items left in the file?

**Review by emlowe — CHANGES_REQUESTED** [review]

File: chia/wallet/nft_wallet/nft_wallet.py [hunk 2]
```diff
@@ -782,7 +794,10 @@ async def make_nft1_offer(
         fee: uint64,
         min_coin_amount: Optional[uint64] = None,
         max_coin_amount: Optional[uint64] = None,
+        old: bool = False,
     ) -> Offer:
+        DESIRED_OFFER_MOD = OFFER_MOD_OLD if old else OFFER_MOD
```
Comment (emlowe) [inline chia/wallet/nft_wallet/nft_wallet.py:799]: any thought about wrapping MOD and MOD_HASH into a Class or other structure in order to keep them together?

File: chia/wallet/trading/offer.py [hunk 3]
```diff
@@ -539,7 +547,11 @@ def from_spend_bundle(cls, bundle: SpendBundle) -> Offer:
         requested_payments: Dict[Optional[bytes32], List[NotarizedPayment]] = {}
         driver_dict: Dict[bytes32, PuzzleInfo] = {}
         leftover_coin_spends: List[CoinSpend] = []
+        old: bool = False
         for coin_spend in bundle.coin_spends:
+            if not old and OFFER_MOD_OLD_BYTES in bytes(coin_spend):
```
Comment (emlowe) [inline chia/wallet/trading/offer.py:552]: This seems the only place this is used - can you just do bytes() call here

**Review by emlowe — CHANGES_REQUESTED** [review]

File: chia/wallet/nft_wallet/nft_wallet.py (hunk 2 above)
Comment (Quexington) [reply chia/wallet/nft_wallet/nft_wallet.py:794]: Can you elaborate on the benefit of this? Or is this just a general thought?

Comment (emlowe) [reply chia/wallet/nft_wallet/nft_wallet.py:794]: Mostly a general thought - you can't mix up the two and you don't need to set them individually. I was mostly thinking it avoids a class of possible typos where you use the wrong hash

Comment (emlowe) [reply chia/wallet/nft_wallet/nft_wallet.py:794]: a NamedTuple maybe

Comment (altendky) [reply chia/wallet/nft_wallet/nft_wallet.py:794]: `namedtuple` is, at least in my mind, primarily a migration tool for moving from regular `tuple` to a full class, presumably a dataclass.  It provides both the old indexing interface that you would be deprecating and the new attribute access you want to move towards.  But I don't think that applies here.

Comment (altendky) [reply chia/wallet/nft_wallet/nft_wallet.py:794]: This sounds maybe like the `NewType` stuff that Adam is playing around with.

**Review by emlowe — DISMISSED** [review]

File: chia/wallet/nft_wallet/nft_wallet.py (hunk 2 above)
Comment (emlowe) [reply chia/wallet/nft_wallet/nft_wallet.py:794]: this isn't that critical so resolving conversation

**Review by arvidn — APPROVED** [review]

**Review by emlowe — APPROVED** [review]

**Pull request merged**